### PR TITLE
fix(frontend):  SQL server 集群管理 #2996

### PR DESCRIPTION
**Pull request by styLjc**

### CODE REVIEW

**Comment by tencentblueking-adm** [comment]
[CLA assistant check](https://bk-cla.bktencent.com:443/TencentBlueKing/blueking-dbm?pullRequest=3201) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bk-cla.bktencent.com:443/TencentBlueKing/blueking-dbm?pullRequest=3201) before we can accept your contribution.<br/><hr/>**v_ljcgliu(刘江城)** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bk-cla.bktencent.com:443/check/TencentBlueKing/blueking-dbm?pullRequest=3201) it.</sub>

File: dbm-ui/frontend/src/views/service-apply/create-db-module/SqlServerCreateDbModule.vue [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+    },
+    {
+      value: 'alwaysOn',
+      label: 'always on',
+    },
+  ];
+
+  const characterSets = ['Chinese_PRC_CI_AS', 'Latin1_General_100_CI_AS'];
+
+  const bizInfo = bizs.find(info => info.bk_biz_id === currentBizId) || { name: '' };
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/service-apply/create-db-module/SqlServerCreateDbModule.vue:199]: 这个不用computed没问题吗？

File: dbm-ui/frontend/src/views/service-apply/create-db-module/SqlServerCreateDbModule.vue [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+      if (res.db_module_id) {
+        messageSuccess(t('创建DB模块并绑定数据库配置成功'));
+        window.changeConfirm = false;
+        router.go(-1);
+      }
+    },
+  });
+
+  const {
+    run: runSaveModuleDeploy,
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/service-apply/create-db-module/SqlServerCreateDbModule.vue]: 一个可以不用换行

File: dbm-ui/frontend/src/views/service-apply/create-db-module/SqlServerCreateDbModule.vue [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+  const getSmartActionOffsetTarget = () => document.querySelector('.bk-form-content');
+
+  /**
+   * 提交表单
+   */
+  const handleSubmit = async () => {
+    const clusterType = String(route.query.cluster_type);
+    const bizId = Number(route.params.bk_biz_id);
+    // 校验表单信息
+    await createModuleFormRef.value.validate();
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/service-apply/create-db-module/SqlServerCreateDbModule.vue:268]: // 校验表单信息
await createModuleFormRef.value.validate();
const clusterType = String(route.query.cluster_type);
const bizId = Number(route.params.bk_biz_id);

File: dbm-ui/frontend/src/views/service-apply/create-db-module/SqlServerCreateDbModule.vue [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+        :disabled="isLoading"
+        @click="resetFormData()">
+        {{ t('重置') }}
+      </BkButton>
+    </template>
+  </SmartAction>
+</template>
+
+<script setup lang="ts">
+  import { merge } from 'lodash';
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/service-apply/create-db-module/SqlServerCreateDbModule.vue]: lodash统一用 import _ from 'lodash'

File: dbm-ui/frontend/src/views/sqlserver-manage/apply/SqlServer.vue [hunk 5]
```diff
@@ -0,0 +1,723 @@
...
+      {
+        message: t('DB模块名不能为空'),
+        trigger: 'blur',
+        validator: (val: number) => !!val,
+      },
+    ],
+    'details.nodes.backend': [{
+      message: t('请添加服务器'),
+      trigger: 'change',
+      validator: () => {
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/apply/SqlServer.vue]: validator: () => formData.details.nodes.backend.length !== 0

File: dbm-ui/frontend/src/views/sqlserver-manage/apply/components/DomainTable.vue [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+        width: 60,
+      },
+      {
+        label: () => (
+        <span>
+          {
+            props.isSqlserverSingle ? t('域名') : t('主域名')
+          }
+          {
+            tableData.value.length !== 0
```
Comment (3octaves) [inline dbm-ui/frontend/src/views/sqlserver-manage/apply/components/DomainTable.vue]: 这种统一为如下格式，否则换行很奇怪：
条件 && （
<BatchEdit />
）

File: dbm-ui/frontend/src/views/sqlserver-manage/apply/components/DomainTable.vue [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+  };
+
+  /**
+   * 渲染域名编辑
+   */
+  const renderDomain = (
+    rowIndex: number,
+    isMain = false,
+  ) => (
+  <div class="domain-address">
```
Comment (3octaves) [inline dbm-ui/frontend/src/views/sqlserver-manage/apply/components/DomainTable.vue]: 注意缩进

File: dbm-ui/frontend/src/views/sqlserver-manage/apply/SqlServer.vue [hunk 8]
```diff
@@ -0,0 +1,723 @@
...
+  };
+
+  /**
+   * 变更所属管控区域
+   */
+  const handleChangeCloud = (info: {
+    id: number | string,
+    name: string
+  }) => {
+    cloudInfo.value.id = info.id;
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/apply/SqlServer.vue]: cloudInfo.value  = info

File: dbm-ui/frontend/src/views/sqlserver-manage/apply/SqlServer.vue [hunk 9]
```diff
@@ -0,0 +1,723 @@
...
+   * 变更业务选择
+   */
+  const handleChangeBiz = () => {
+    formData.details.db_module_id = 0;
+    formData.details.nodes.backend = [];
+    moduleRef.value.clearValidate();
+  };
+
+  // 获取 DM模块
+  watch(route.query, () =>  getModulesConfig(), {
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/apply/SqlServer.vue]: 多了空格

File: dbm-ui/frontend/src/views/sqlserver-manage/apply/components/BatchEdit.vue [hunk 10]
```diff
@@ -0,0 +1,278 @@
...
+  <BkPopover
+    v-model:is-show="stateShow"
+    :boundary="body"
+    class="batch-edit"
+    theme="light"
+    trigger="manual"
+    :width="540">
+    <i
+      class="db-icon-bulk-edit batch-edit-trigger"
+      @click="() =>stateShow = true" />
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/apply/components/BatchEdit.vue]: 空格

File: dbm-ui/frontend/src/views/sqlserver-manage/apply/components/PreviewTable.vue [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+  interface previewDataType{
+    domain:string,
+    slaveDomain:string,
+    disasterDefence:string,
+    deployStructure:string,
+    version:string,
+    charset:string,
+  }
+
+  interface Props {
```
Comment (3octaves) [inline dbm-ui/frontend/src/views/sqlserver-manage/apply/components/PreviewTable.vue:16]: 这些属性的类型不用分开定义，后续若有用到，直接使用类型推导来获取

File: dbm-ui/frontend/src/views/sqlserver-manage/apply/components/DomainTable.vue (hunk 6 above)
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/apply/components/DomainTable.vue]: tableData.value.length !== 0 &&  \n

File: dbm-ui/frontend/src/views/sqlserver-manage/apply/components/DomainTable.vue [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+  const columns = computed(() => {
+    const columns: Column[] = [
+      {
+        type: 'index',
+        label: t('序号'),
+        width: 60,
+      },
+      {
+        label: () => (
+        <span>
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/apply/components/DomainTable.vue]: 空格

File: dbm-ui/frontend/src/views/sqlserver-manage/apply/components/DomainTable.vue [hunk 13]
```diff
@@ -0,0 +1,244 @@
...
+   * 渲染域名编辑
+   */
+  const renderDomain = (
+    rowIndex: number,
+    isMain = false,
+  ) => (
+  <div class="domain-address">
+    <span>{ props.moduleName }{ isMain ? 'db.' : 'dr.' }</span>
+    {
+      isMain
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/apply/components/DomainTable.vue]: isMain ? ( \n

File: dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/Index.vue [hunk 14]
```diff
@@ -0,0 +1,25 @@
+<template>
+  <StretchLayout
+    :min-left-width="368"
+    name="sqlserverHaClusterList">
+    <template #list>
+      <List
+        v-model:haClusterData="haClusterData" />
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/Index.vue]: 一个属性不换行

File: dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/components/List.vue [hunk 15]
```diff
@@ -0,0 +1,503 @@
...
+          <div style="display: flex; align-items: center;">
+            <db-icon
+              type="copy"
+              v-bk-tooltips={ t('复制主访问入口') }
+              onClick={ () => copy(data.master_enter) }/>
+            <db-icon
+              type="link"
+              v-bk-tooltips={ t('新开tab打开') }/>
+            <div class="text-overflow" v-overflow-tips>
+              {
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/components/List.vue:273]: 功能缺失

File: dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/components/List.vue [hunk 16]
```diff
@@ -0,0 +1,503 @@
...
+    {
+      label: t('管控区域'),
+      field: 'control_area',
+    },
+    {
+      label: t('状态'),
+      field: 'status',
+      sort: true,
+      render: ({ data }: { data: SqlServerClusterModel }) => {
+        const { text, theme } = data.dbStatusConfigureObj;
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/components/List.vue:303]: 换行

File: dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/components/List.vue [hunk 17]
```diff
@@ -0,0 +1,503 @@
...
+        return <DbStatus theme={ theme }>{ text }</DbStatus>;
+      },
+    },
+    {
+      label: t('实例'),
+      field: 'instance_name',
+      render: ({ data }: { data: SqlServerClusterModel }) => (
+      <RenderInstances
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/components/List.vue]: 空格

File: dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/components/List.vue [hunk 18]
```diff
@@ -0,0 +1,503 @@
...
+      field: 'create_time',
+      sort: true,
+    },
+    {
+      label: t('操作'),
+      field: 'operation',
+      width: tableOperationWidth.value,
+      fixed: 'right',
+      render: ({ data }: { data: SqlServerClusterModel }) => (
+      <>
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/components/List.vue]: 空格

File: dbm-ui/frontend/src/views/sqlserver-manage/single-cluster/components/List.vue [hunk 19]
```diff
@@ -0,0 +1,497 @@
...
+              type="copy"
+              v-bk-tooltips={ t('复制主访问入口') }
+              onClick={ () => copy(data.master_enter) } />
+            <db-icon
+              type="link"
+              v-bk-tooltips={ t('新开tab打开') } />
+            <div class="text-overflow" v-overflow-tips>
+              {
+                data.isNew
+                && <span
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/single-cluster/components/List.vue]: 同上

File: dbm-ui/frontend/src/views/sqlserver-manage/single-cluster/components/List.vue [hunk 20]
```diff
@@ -0,0 +1,497 @@
...
+    bk_instance_id: number,
+  }
+
+  const singleClusterData = defineModel<{
+    clusterId: number,
+  }>('singleClusterData');
+
+  const router = useRouter();
+  const route = useRoute();
+  const globalBizsStore = useGlobalBizs();
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/single-cluster/components/List.vue]: 解构当前业务ID

File: dbm-ui/frontend/src/views/sqlserver-manage/single-cluster/components/detail/BaseInfo.vue [hunk 21]
```diff
@@ -0,0 +1,91 @@
...
+    singleClusterData:{
+      clusterId: number
+    }
+  }
+
+  const props = defineProps<Props>();
+
+  const { t } = useI18n();
+
+  const columns:InfoColumn[][] = [
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/single-cluster/components/detail/BaseInfo.vue]: 空格，自查

**Pull request closed without merging**